x-pack/plugins/task_manager/server/lib/set_claim_strategy.ts

```diff
@@ -68,13 +68,32 @@ export function setClaimStrategy(opts: SetClaimStrategyOpts): TaskManagerConfig
     return {
       ...opts.config,
       claim_strategy: CLAIM_STRATEGY_MGET,
-      poll_interval: MGET_DEFAULT_POLL_INTERVAL,
+      poll_interval: maybeSetDefaultInterval(opts.config, MGET_DEFAULT_POLL_INTERVAL),
     };
   }
 
   return {
     ...opts.config,
     claim_strategy: CLAIM_STRATEGY_UPDATE_BY_QUERY,
-    poll_interval: DEFAULT_POLL_INTERVAL,
+    poll_interval: maybeSetDefaultInterval(opts.config, DEFAULT_POLL_INTERVAL),
   };
 }
+
+/**
+ * If the poll interval is not overridden by the user, return the specified default.
+ *
+ * @param config Current config
+ * @param defaultPollInterval Default to set
+ */
+function maybeSetDefaultInterval(config: TaskManagerConfig, defaultPollInterval: number) {
```

Comment:
Thanks for the PR! I was looking at how this function differs to letting the `poll_interval` overwrite completely (`opts.config.poll_interval ?? DEFAULT_POLL_INTERVAL`).

From looking at the function, it seems if ever the user sets `xpack.task_manager.poll_interval: 3000` in their yml and their deployment is auto enrolled to the `mget` task claiming strategy, we're going to discard the `3000` and set `500` as the value. I guess this is the current behaviour on 8.16, was your intent to preserve this as well?

Reply:
IIUC, this method returns the default only if the user didn't provide it (which, in case they didn't set "mget" *in their `kibana.yml`* is DEFAULT_POLL_INTERVAL). 

I don't think that the a/b decision logic in this file affects it (at least, that was not my intention).

In case they set "mget" in their kibana.yml, this method is not called because there's an early return at the top of this file.